src/marqo/tensor_search/validation.py

```diff
@@ -482,3 +527,20 @@ def validate_delete_docs_request(delete_request: MqDeleteDocsRequest, max_delete
         validate_id(_id)
 
     return delete_request
+
+
+def validate_result_count(result_count: int):
+    """Validates the result count (limit) for a search operation.
+
+    Args:
+        result_count: the result count to validate
+    Returns:
+        result_count, if nothing is raised
+    """
+    if not isinstance(result_count, int):
```

Comment:
You don't need type validation since these values pass through pydantic models and pydantic takes care of that